lib/phase/functional.js

```diff
@@ -181,14 +181,17 @@ function validateWorkflow(doc) {
 module.exports = flattenedDoc => (
     new Promise((resolve, reject) => {
         const doc = flattenedDoc;
+        const hasRequires = Object.keys(doc.jobs).some(jobName => doc.jobs[jobName].requires);
         let errors = [];
 
         // Jobs
         errors = errors.concat(validateJobSchema(doc));
 
         // Workflow
```

Comment:
Maybe it'd be helpful to have comments like `legacy workflow` or something for code that we'll need to remove

Reply:
Added some comments